htdocs/product/stock/massstockmove.php

```diff
@@ -336,11 +336,11 @@
 					continue;
 				}
 				//var_dump($data);
-				$tmp_id_sw = $data[$i][0]['val'];
-				$tmp_id_tw = $data[$i][1]['val'];
-				$tmp_id_product = $data[$i][2]['val'];
-				$tmp_qty = $data[$i][3]['val'];
-				$tmp_batch = $data[$i][4]['val'];
+				$tmp_id_sw = dol_escape_htmltag($data[$i][0]['val']);
```

Comment:
No. In memory, we must have date as a brute data.
The escapement must be done when we need it. So if we need to output the data for sql, we will escape using db->escape, if we need to use it for output on screen, we will use dol_escape_htmltag but this will be done during the print of the html content.
Imagine that the data $tmp_batch is used later for a test or to be written into a file, you will write a value with html output escapement when you don't want.

Data in memory must be brut and utf8 encoded.
Escapement must be done when we need it (the later as possible).